pkg/ipam/metadata/manager.go

```diff
@@ -95,11 +96,34 @@ func splitK8sPodName(owner string) (namespace, name string, ok bool) {
 	return namespace, name, true
 }
 
-func (m *Manager) GetIPPoolForPod(owner string) (pool string, err error) {
+func determinePoolByAnnotations(ipv4Annotation, ipv6Annotation, ipAnnotation string, family ipam.Family) (pool string, ok bool) {
```

Comment:
nit: Could you pass in `annotations map[string]string` as the argument here, and then populate it either from `pod.Annotations` or `namespace.Annotations`?

Reply:
OK, I will change it. Thanks!